phac-nml/irida-next
# Feat: Sample::Query form object

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR refactors the Samples table to use ActiveModel to create a better search form for the Samples table, following this [guide](https://thoughtbot.com/blog/rails-search-form-tutorial).

This PR removes initial implementation of metadata searching as that will be reimplemented using SearchKick.

This is a prerequisite for changing to SearchKick.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Launch server
2. Interact with Project and Groups Samples tables and ensure everything works as expected. The system tests should catch all cases.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
